### PR TITLE
implement partialord min/max

### PR DESCRIPTION
**Pull request by sebasv**
Implements #512 , but for `PartialOrd`. Unorderable cases such as `f64::NAN` result in an early return of `None`. Ideally there is also a variant that ignores unorderable cases, something like `scalar_max_ignore_unorderable()`, but with a more sensible name.

### CODE REVIEW

**Comment by sebasv** [comment]
Forgot to mention! This depends on #507 . Travis should pass when that PR makes it into master.

File: src/numeric/impl_numeric.rs [hunk 1]
```diff
@@ -46,6 +46,84 @@ impl<A, S, D> ArrayBase<S, D>
...
+    /// 
+    /// let a = arr2(&[[1., 2.], [3., 4.]]);
+    /// 
+    /// assert_eq!(a.scalar_max(), Some(&4.));
+    /// 
+    /// let b = arr2(&[[1., f64::NAN], [3., 4.]]);
+    /// 
+    /// assert_eq!(b.scalar_max(), None);
+    /// ```
+    pub fn scalar_max(&self) -> Option<&A>
```
Comment (bluss) [inline src/numeric/impl_numeric.rs]: Any reason this is not named max?

Comment (bluss) [reply src/numeric/impl_numeric.rs]: Last I knew the plan was to rename scalar_sum to sum, so don't want it to be followed. That said, last I knew was some time ago. Commit b75d438daea2d0c5ce684fed1a5a1ebde7d4e166

Comment (sebasv) [reply src/numeric/impl_numeric.rs]: Excellent, renamed!

**Comment by bluss** [comment]
Needs discussion what to do with the overlap with ndarray-stats https://docs.rs/ndarray-stats/

If this is in ndarray, I'd prefer something that's more geared towards the needs of floating point elements

**Comment by thecooldrop** [comment]
Hey @bluss what do you mean, when you say that it should be more geared toward floating point elements?

**Comment by sebasv** [comment]
Closing as I don't see myself picking this up anytime soon. Feel free to take over

**Pull request closed without merging**